ocaml/xapi/xapi_pgpu.ml

```diff
@@ -307,3 +354,25 @@ let disable_dom0_access ~__context ~self =
   if not (Pool_features.is_enabled ~__context Features.Integrated_GPU)
   then raise Api_errors.(Server_error (feature_restricted, []));
   update_dom0_access ~__context ~self ~action:`disable
+
+(* This must be run LOCALLY on the host that is about to start a VM that is
+ * going to use a vgpu backed by an AMD MxGPU pgpu. *)
+let mxgpu_vf_setup ~__context =
+  let localhost = Helpers.get_localhost ~__context in
+  (* From the modprobe(8) manpage:
+   * --first-time
+   *     Normally, modprobe will succeed (and do nothing) if told to insert a
+   *     module which is already present or to remove a module which isn't
+   *     present. This is ideal for simple scripts; however, more complicated
+   *     scripts often want to know whether modprobe really did something: this
+   *     option makes modprobe fail in the case that it actually didn't do
+   *     anything. *)
+  ignore (Forkhelpers.execute_command_get_output !Xapi_globs.modprobe_path ["gim"]);
+  (* Update the gpus even if the module was present already, in case it was
+   * already loaded before xapi was (re)started. *)
+  Xapi_pci.update_pcis ~__context ~host:localhost;
+  (* Potential optimisation: make update_pcis return a value telling whether
+   * it changed anything, and stop here if it did not. *)
+  List.iter
```

Comment:
It would be harmless, but would do a fair bit of needless work, and doing it this way is still quite simple.
There will never be any need to create or update any pgpu DB objects here: all we want to do is create a pci DB objects for each VF pci device, and set its `physical_function` field (which will automatically cause the `virtual_functions` field of its PF pci DB object to be updated).